Signal/src/ViewControllers/ContactsViewHelper.m

```diff
@@ -26,19 +26,26 @@ @interface ContactsViewHelper ()
 
 @property (nonatomic) NSArray<NSString *> *blockedPhoneNumbers;
 
+@property (nonatomic) BOOL suppressDelegateEvents;
+
 @end
 
 #pragma mark -
 
 @implementation ContactsViewHelper
 
-- (instancetype)init
+- (instancetype)initWithDelegate:(id<ContactsViewHelperDelegate>)delegate
 {
     self = [super init];
     if (!self) {
         return self;
     }
 
+    OWSAssert(delegate);
+    _delegate = delegate;
+
+    self.suppressDelegateEvents = YES;
```

Comment:
We don't want to fire delegate events during initialization, but we do want to consult the "should hide local number" delegate method.